recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,238 @@
+        "discovery_semaphore": [True, False],
+        "discovery_multicast": [True, False],
+        "query": [True, False],
+        "encription": [True, False],
+        "json_support": [True, False],
+        "pub_sub": ["None", "Simple", "Ethernet", "Ethernet_XDP"],
+        "data_access": [True, False],
+        "compiled_nodeset_descriptions": [True, False],
+        "namescpae_zero": ["MINIMAL", "REDUCED", "FULL"],
+        "embedded_profile": [True, False],
+        "typenames": [True, False],
+        "hardening": [True, False],
+        "cpp_compatible": [True, False],
+        "readable_statuscodes": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": True,
+        "historize": False,
+        "logging_level": "Info",
+        "subscription": True,
+        "subscription_events": False,
+        "methods": True,
+        "dynamic_nodes": True,
+        "single_header": False,
+        "multithreading": False,
+        "imutable_nodes": False,
+        "web_socket": False,
+        "discovery": True,
+        "discovery_semaphore": True,
+        "discovery_multicast": False,
+        "query": False,
+        "encription": False,
+        "json_support": False,
+        "pub_sub": "None",
+        "data_access": True,
+        "compiled_nodeset_descriptions": True,
+        "namescpae_zero": "FULL",
+        "embedded_profile": False,
+        "typenames": True,
+        "hardening": False,
+        "cpp_compatible": False,
+        "readable_statuscodes": True
+    }
+    generators = "cmake", "cmake_find_package"
+
+    def requirements(self):
+        if self.options.encription:
+            self.requires("mbedtls/2.16.3-gpl")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _download_deps(self, zip_link, dep_name):
+        zip_name = dep_name + ".zip"
+        download(zip_link, zip_name)
+        dep_path = self._source_subfolder + "/deps/" + dep_name
+        unzip(zip_name)
+        full_path = str(urlparse(zip_link).path).split("/")
+        dir_name = full_path[2] + "-" + full_path[4]
+        dir_name = dir_name[:-4]
+        files = os.listdir(dir_name)
+        for f in files:
+            shutil.move(os.path.join(dir_name, f), dep_path)
+        os.unlink(zip_name)
+        shutil.rmtree(dir_name)
+
+    def _download_ua_nodes(self):
+        link = ""
+        if self.version == "1.0.0":
+            link = "https://github.com/OPCFoundation/UA-Nodeset/archive/0777abd1bc407b4dbd79abc515864f8c3ce6812b.zip"
+        elif self.version == "1.0.1":
+            link = "https://github.com/OPCFoundation/UA-Nodeset/archive/0777abd1bc407b4dbd79abc515864f8c3ce6812b.zip"
+        self._download_deps(zip_link=link, dep_name="ua-nodeset")
+
+    def _download_mdnsd(self):
+        link = ""
+        if self.version == "1.0.0":
+            link = "https://github.com/Pro/mdnsd/archive/8fe3a7e7e9d0a9196b126c64f0d1905569b83d40.zip"
```

Comment:
Likewise. Use Conan packages or add to conandata.yml.

Reply:
[mdnsd](https://github.com/Pro/mdnsd] is not a Conan dependency

Reply:
Having all dependencies as conan recipes/packages would be the correct way...
I don't think the maintainers will accept this recipe if it fetches its dependencies as git submodules.
mdnsd might be useful for somebody else..

Reply:
alright, I'll develop those recipes as well, but until they are accepted and merged in, I'll keep this method, so I can continue to test on the central CI/CD pipeline. I'll upload a git diff for this method as well as links to the new recipes in this conversation.

Reply:
Great!

Reply:
As a first step, I asked for permission from the author [here](https://github.com/Pro/mdnsd/issues/17] if he agrees I will start development on the recipe.

Reply:
I agree :+1: Thanks for your effort!

Reply:
Just asking, isn't https://github.com/troglobit/mdnsd the upstream mdnsd?

Reply:
Yes, it was the original upstream repo. But at the time I integrated mDNS into open62541, that upstream repo was not maintained anymore. In my fork I fixed many memory issued and made it compilable for windows.
I guess the upstream repo is therefore currently not compatible with my fork.
At some point I need to merge these two again as it seems the upstream is active again.

Reply:
first draft of [mdnsd](https://github.com/conan-io/conan-center-index/pull/2009) has been published as a PR! 

I am not 100% sure if the [test_package.cpp](https://github.com/conan-io/conan-center-index/blob/edf773c666424a22bcda9745a64eea9a835dcbb7/recipes/mdnsd/all/test_package/test_package.cpp) is the best implementation since I am not familiar with this library, if you have any ideas on how to improve it, please let me know!

Reply:
Unfortunately, my initial efforts to contribute all the necessary changes back to upstream were not really handled in a good way by the maintainer:

troglobit/mdnsd#1

Windows support was ignored at all.

Therefore I continued in my own fork. Checked yesterday how much effort it would be to create a new PR, but the fork already diverged too much.